open_optical_gating/cli/cli.py

```diff
@@ -357,22 +393,28 @@ def pog_state(self, frame):
             self.frame_history[-1, -1],
         )
 
-        # If at least one period has passed, have a go at predicting a trigger
+        # If at least one period has passed, have a go at predicting a future trigger time
         if self.frame_num - 1 > self.pog_settings["referencePeriod"]:
             logger.debug("Predicting trigger...")
 
+            # TODO: JT writes: this seems as good a place as any to highlight the general issue that the code is not doing a great job of precise timing.
+            # It determines a delay time before sending the trigger, but then executes a bunch more code.
+            # Oh and, more importantly, that delay time is then treated relative to “current_time”, which is set *after* doing the phase-matching.
+            # That is going to reduce accuracy and precision, and also makes me even more uncomfortable in terms of future-proofing.
+            # I think it would be much better to pass around absolute times, not deltas.

```

Comment:
Good point re extra code after deteriming time. There was a reason, three years ago, to use relative times... but it could just have been because that was of interest to me in my offline analyses?

Reply:
Raised issue; closing this conversation